examples/end-to-end-examples/support_chatbot/app_demo.py

```diff
@@ -0,0 +1,208 @@
+                     Provide a helpful and simple answer given the following context to the question. If you do not know, just say "I 
+                     do not know".<|im_end|>
+                     <|im_start|>context
+                     {context}<|im_end|>
+                     <|im_start|>user
+                     {question}<|im_end|>
+                     <|im_start|>assistant""")
+
+def parse_args() -> Namespace:
+    """Parse commandline arguments."""
+    parser = ArgumentParser(
+        description=
+        'Run a chatbot!'
+    )
+    parser.add_argument(
+        '--endpoint_url',
+        type=str,
+        default='https://models.hosted-on.mosaicml.hosting/mpt-30b-chat/v1/predict',
+        required=False,
+        help='The endpoint of our MosaicML LLM Model')
+    parser.add_argument(
+        '--model_name',
+        type=str,
+        default='mpt-30b-chat',
+        required=False,
+        help='only evals offered as of now are mpt-30b-chat and mpt-7b')
+    parser.add_argument(
+        '--max_length',
+        type=int,
+        default=5000,
+        required=False,
+        help='The maximum size tokens in model')
+    parser.add_argument(
+        '--chunk_size',
+        type=int,
+        default=800,
+        required=False,
+        help='The chunk size when splitting documents')
+    parser.add_argument(
+        '--chunk_overlap',
+        type=int,
+        default=400,
+        required=False,
+        help='The overlap between chunks when splitting documents')
+    parser.add_argument(
+        '--retrieval_k',
+        type=int,
+        default=5,
+        required=False,
+        help='The number of chunks to retrieve as context from vector store')
+    parser.add_argument(
+        '--model_k',
+        type=int,
+        default=10,
+        required=False,
+        help='The number of outputs model should output')
+    parser.add_argument(
+        '--repository_urls',
+        type=str,
+        nargs='*',  
+        default=['https://github.com/mosaicml/composer',
+                 'https://github.com/mosaicml/streaming',
+                 'https://github.com/mosaicml/examples',
+                 'https://github.com/mosaicml/diffusion',
+                 'https://github.com/mosaicml/llm-foundry'], 
+        required=False,
+        help='The GitHub repository URLs to download'
+    )
+    parser.add_argument(
+        '--complex_data_dir',
+        type=str,
+        required=False,
+        help='complex eval data for human eval')
+    parser.add_argument(
+        '--simple_data_dir',
+        type=str,
+        required=False,
+        help='simple eval data for string comparison')
+    parser.add_argument(
+        '--complex_chat',
+        type=int,
+        default=0,
+        required=False,
+        help='Whether to use subquery chatting')
+    
+    parsed = parser.parse_args()
+    
+    if parsed.repository_urls is not None:
```

Comment:
still needed because im using '*' not '+', so it can be empty. I think this is fine because we shouldn't have to force the user to provide a git link if the vectors already exist

Reply:
got it, sg